### PR TITLE
Internal improvement: Make source-fetcher throw on invalid network

### PR DESCRIPTION
**Pull request by haltman-at**
**Note: This PR is a breaking change to `@truffle/source-fetcher`**.

I realized that source-fetcher probably ought to throw if you give it an invalid network ID.  So now it does.  The `isNetworkValid` method is gone, because it's now unnecessary.  Note obviously this required appropriate modifications to `external.js`.

### CODE REVIEW

File: packages/core/lib/debug/external.js [hunk 1]
```diff
@@ -45,27 +33,32 @@ class DebugExternalHandler {
         }
       }
     } else {
-      sortedFetchers = allFetchers;
+      sortedFetchers = Fetchers;
     }
-    //to get the final list, we'll filter out ones that don't support this
+    const networkId = await new Web3(this.config.provider).eth.net.getId(); //note: this is a number
```
Comment (gnidan) [inline packages/core/lib/debug/external.js:38]: BTW you don't need to do this. This is available on the config:
```suggestion
    const networkId = this.config.network_id; //note: this is a number
```

**Review by gnidan — CHANGES_REQUESTED** [review]
Looks good apart from this one thing

File: packages/core/lib/debug/external.js (hunk 1 above)
Comment (haltman-at) [reply packages/core/lib/debug/external.js:38]: Oh whoa. We've always been able to do this...?

Comment (gnidan) [reply packages/core/lib/debug/external.js:38]: Yeah, found it yesterday!

**Comment by haltman-at** [comment]
Actually wait I want to make another change to this too, you'll see it in a moment...

**Comment by haltman-at** [comment]
OK, I fixed that, and also I exported the error so I can do a proper error check with `instanceof` instead of relying on `error.name`.  For some reason I got the mistaken idea that would be somehow messy so didn't do that the first time around, but it's not.

**Review by gnidan — APPROVED** [review]
Looks good!

**Pull request merged**